test/gtest/ucp/test_ucp_am.cc

```diff
@@ -816,7 +816,7 @@ class test_ucp_am_nbx_closed_ep : public test_ucp_am_nbx {
                                 bool rx_expected = false)
     {
         skip_loopback();
-        test_am_send_recv(0, max_am_hdr()); // warmup wireup
+        test_am_send_recv(0, 0); // warmup wireup
```

Comment:
why needed in this PR?

Reply:
so AFAIU this PR handles 2 things, right?
1. add CHECK flag for AM and implement it with shared memory
2. fix failure in AM (with next KA PR) test due to late resolve of remote id

Reply:
> so AFAIU this PR handles 2 things, right?
> 
> 1. add CHECK flag for AM and implement it with shared memory
> 2. fix failure in AM (with next KA PR) test due to late resolve of remote id

yes, correct.
but the 2nd item discovers the issue from the 1st item.